cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -671,6 +671,18 @@ if (ARROW_WITH_ZLIB)
 # ZLIB
 
   if("${ZLIB_HOME}" STREQUAL "")
+    if(ARROW_ZLIB_VENDORED)
```

Comment:
IMHO, we should make it an error to set both ZLIB_HOME and ARROW_ZLIB_VENDORED. It's already annoying enough to debug configuration issues.

Reply:
I've removed `ARROW_ZLIB_VENDORED` to use system zlib by default and vendored zlib as fallback.